internal/warehouse/githubActions.go

```diff
@@ -0,0 +1,383 @@
+			Repoid:         repoID,
+			ID:             *workflow.ID,
+			Workflownodeid: *workflow.NodeID,
+			Name:           *workflow.Name,
+			Path:           *workflow.Path,
+			State:          *workflow.State,
+			Createdat:      workflow.CreatedAt.Time,
+			Updatedat:      workflow.UpdatedAt.Time,
+			Url:            *workflow.URL,
+			Htmlurl:        *workflow.HTMLURL,
+			Badgeurl:       *workflow.BadgeURL,
+		}); err != nil {
+			return err
+		}
+	}
+
+	return tx.Commit(ctx)
+}
+
+func (w *warehouse) handleUpsertWorkflowRuns(ctx context.Context, workflowRunPage []*github.WorkflowRun, repoID uuid.UUID) error {
+	var tx pgx.Tx
+	var err error
+
+	if tx, err = w.pool.BeginTx(ctx, pgx.TxOptions{}); err != nil {
+		return fmt.Errorf("begin tx: %w", err)
+	}
+
+	defer func() {
+		if err := tx.Rollback(ctx); err != nil {
+			if !errors.Is(err, pgx.ErrTxClosed) {
+				w.logger.Err(err).Msgf("could not rollback transaction")
+			}
+		}
+	}()
+
+	for _, workflowRun := range workflowRunPage {
+
+		pullRequestsBytes, err := json.Marshal(&workflowRun.PullRequests)
+
+		if err != nil {
+			return err
+		}
+
+		headCommitBytes, err := json.Marshal(&workflowRun.HeadCommit)
+
+		if err != nil {
+			return err
+		}
+
+		if err := w.db.WithTx(tx).UpserWorkflowRuns(ctx, db.UpserWorkflowRunsParams{
+			RepoID:            repoID,
+			ID:                *workflowRun.ID,
+			Workflowrunnodeid: *workflowRun.NodeID,
+			Name:              *workflowRun.Name,
+			Headbranch:        *workflowRun.HeadBranch,
+			Runnumber:         int32(*workflowRun.RunNumber),
+			Runattemp:         int32(*workflowRun.RunAttempt),
+			Event:             *workflowRun.Event,
+			Status:            *workflowRun.Status,
+			Conclusion:        *workflowRun.Conclusion,
+			Workflowid:        *workflowRun.WorkflowID,
+			Checksuiteid:      *workflowRun.CheckSuiteID,
+			Checksuitenodeid:  *workflowRun.CheckSuiteNodeID,
+			Url:               *workflowRun.URL,
+			Htmlurl:           *workflowRun.HTMLURL,
+			Pullrequest: pgtype.JSONB{
+				Status: pgtype.Present,
+				Bytes:  pullRequestsBytes,
+			},
+			Createdat:     workflowRun.CreatedAt.Time,
+			Updatedat:     workflowRun.UpdatedAt.Time,
+			Runstartedat:  workflowRun.RunStartedAt.Time,
+			Jobsurl:       *workflowRun.JobsURL,
+			Logsurl:       *workflowRun.LogsURL,
+			Checksuiteurl: *workflowRun.CheckSuiteURL,
+			Artifactsurl:  *workflowRun.ArtifactsURL,
+			Cancelurl:     *workflowRun.CancelURL,
+			Rerunurl:      *workflowRun.RerunURL,
+			Headcommit: pgtype.JSONB{
+				Status: pgtype.Present,
+				Bytes:  headCommitBytes,
+			},
+			Workflowurl:       *workflowRun.WorkflowURL,
+			Repositoryurl:     *workflowRun.Repository.URL,
+			Headrepositoryurl: *workflowRun.HeadRepository.URL,
+		}); err != nil {
+			return err
+		}
+
+	}
+
+	return tx.Commit(ctx)
+}
+
+func (w *warehouse) handleUpsertWorkflowJobs(ctx context.Context, workflowJob *github.WorkflowJob, log string, repoID uuid.UUID) error {
```

Comment:
This is plural but only accepts one job.

Reply:
Done :)